theano/tensor/nnet/tests/test_bn.py

```diff
@@ -6,6 +6,47 @@
 from theano.tensor.nnet.bn import batch_normalization
 
 
+def test_BNComposite():
+    try:
+        orig = theano.config.compute_test_value
+
+        theano.config.compute_test_value = 'raise'
+
+        def bn_ref(x, G, B, M, V):
+            n = (x - M) / V
+            return n * G + B
+
+        numpy.random.seed(1234)
+        X = 1 + numpy.random.random([10, 20]).astype('float32')
+        B = 1 + numpy.random.random([20]).astype('float32')
+        G = 1 + numpy.random.random([20]).astype('float32')
+        M = 1 + numpy.random.random([20]).astype('float32')
+        V = 1 + numpy.random.random([20]).astype('float32')
+
+        x = theano.tensor.matrix('x')
+        b = theano.tensor.vector('b')
+        g = theano.tensor.vector('g')
+        m = theano.tensor.vector('m')
+        v = theano.tensor.vector('v')
+
+        x.tag.test_value = numpy.random.rand(2, 2)
```

Comment:
Cast the value to floatX like value.astype(theano.config.floatX).

Without this, the tests fail when floatX=float32.